### PR TITLE
Migrate interface logic from microsetta-private-api

### PR DESCRIPTION
**Pull request by wasade**
cc @dhakim87
cc biocore/microsetta-private-api#175

- files migrated
- `microsetta_private_api` -> `microsetta_interface`
- added a `private_api_endpoint` to `SERVER_CONFIG`
- basic package pieces like `setup.py` added
- `.yaml` updated to point to `implementation`
- authrocket updated to allow callbacks to a different port
- login and use verified (not tested comprehensively) 
- created a `Source` class stub as an enum to avoid an import from `microsetta_private_api`

### CODE REVIEW

**Comment by dhakim87** [comment]
Approved - (I'm not a maintainer of this repository so I technically can't)

That's a ton of files, but it looks like they were just moved over.  
It looks like you disabled code coverage checks, did you want to turn it back on (No is fine)
Which side did the redis cache end up on?  I think the redis cache was implemented interface side but I don't see it in the readme (I'd have sworn the readme used to have something about starting it up).  (It does properly install it though according to the travis config I think)

**Comment by wasade** [comment]
Thanks, @dhakim87! Code coverage was not disabled, there are just like zero tests. 

Redis cache is here and comes down from conda requirements.

File: microsetta_interface/templates/new_participant.jinja2 [hunk 1]
```diff
@@ -0,0 +1,255 @@
...
+<div id="0-6-div" style='text-align: left;padding: 10px'>
+<div class='scrolltext'>
+{{ tl['CONSENT_YOUR_CHILD'] }}
+{{ tl['PARTICIPATION_AGREEMENT'] }}
+{{ tl['EXHIBIT_A'] }}
+</div>
+    <div class="" id="consent">
+        <hr>
+        <a href="http://oag.ca.gov/sites/all/files/agweb/pdfs/research/bill_of_rights.pdf" target="_blank">{{ tl['BILL_OF_RIGHTS']|e}}</a><br/>
+        <form id="0-6-form" name="consent_info" onsubmit="return validate06();" action="{{ post_url }}" method="post">
```
Comment (dhakim87) [inline microsetta_interface/templates/new_participant.jinja2:146]: Where is {{ post_url }} getting passed in?  It looks like there were changes made during the migration that might have stopped sending this in, only ting I see being passed to jinja2 renderer is the variable tl and the default variables passed to all templates.  Do the post_url links work in the consent doc after the migration?  There should be several of them in this file.

Comment (wasade) [reply microsetta_interface/templates/new_participant.jinja2:146]: Zero changes with what was in microsetta-private-api:

```bash
(microsetta-admin-37) 13:51:26 (dtmcdonald@here):tr$ wget https://raw.githubusercontent.com/biocore/microsetta-private-api/master/microsetta_private_api/templates/new_participant.jinja2
--2020-07-08 13:51:28--  https://raw.githubusercontent.com/biocore/microsetta-private-api/master/microsetta_private_api/templates/new_participant.jinja2
Resolving raw.githubusercontent.com... 151.101.196.133
Connecting to raw.githubusercontent.com|151.101.196.133|:443... connected.
HTTP request sent, awaiting response... 200 OK
Length: 13040 (13K) [text/plain]
Saving to: 'new_participant.jinja2'

new_participant.jinja2                            100%[=============================================================================================================>]  12.73K  --.-KB/s    in 0.04s

2020-07-08 13:51:29 (344 KB/s) - 'new_participant.jinja2' saved [13040/13040]

(microsetta-admin-37) 13:51:29 (dtmcdonald@here):tr$ diff new_participant.jinja2 microsetta-interface/microsetta_interface/templates/new_participant.jinja2
(microsetta-admin-37) 13:51:29 (dtmcdonald@here):tr$
```

Comment (wasade) [reply microsetta_interface/templates/new_participant.jinja2:146]: post_url is coming from the private API in the json object

Comment (dhakim87) [reply microsetta_interface/templates/new_participant.jinja2:146]: Change is in the set of variables passed in to jinja2 rendering:

``` 
    consent_html = render_template(
        "new_participant.jinja2",
        tl=localization_info[localization.NEW_PARTICIPANT_KEY],
        lang_tag=language_tag,
        post_url=consent_post_url
    )
```

```
    return _render_with_defaults('new_participant.jinja2',
                                 tl=consent_output)
```

Comment (wasade) [reply microsetta_interface/templates/new_participant.jinja2:146]: ah, I see now. the page "works" in that i could advance but ill verify the url is there

**Pull request merged**